bench/src/bulk.rs

```diff
@@ -51,27 +62,72 @@ async fn server(mut incoming: quinn::Incoming) -> Result<()> {
     let quinn::NewConnection {
         mut uni_streams, ..
     } = handshake.await.context("handshake failed")?;
-    let mut stream = uni_streams
-        .next()
-        .await
-        .ok_or_else(|| anyhow!("accepting stream failed"))??;
-    trace!("stream established");
+
+    loop {
+        let mut stream = match uni_streams.next().await {
+            None => return Ok(()),
+            Some(Err(quinn::ConnectionError::ApplicationClosed(_))) => return Ok(()),
+            Some(Err(e)) => return Err(e).context("accepting stream failed"),
+            Some(Ok(stream)) => stream,
+        };
+        trace!("stream established");
+
+        let _jh: tokio::task::JoinHandle<Result<()>> = tokio::spawn(async move {
```

Comment:
Why do we need to name the type of the `JoinHandle` here?

Reply:
We have to explicitely write the type so that `?` works. `_jh` can be replaced with `_`. I typically like to still see what gets ignored based on the variable name. But it's obviously shown here by the type parameter. I can remove it.

Reply:
Okay. Maybe also just import `tokio::task::JoinHandle` up top so that we don't have the whole path inline here.